contracts/erc1155/custom/yungwknd/YungWkndGenerative.sol

```diff
@@ -0,0 +1,175 @@
+     * @notice Throw when mint details are queried for a token that hasn't been minted
+     */
+    error InvalidTokenId();
+
+    /**
+     * @notice Data partially used to seed outputs
+     */
+    struct SeedDetails {
+        bytes32 previousBlockHash;
+        uint256 blockTimestamp;
+    }
+
+    /**
+     * @notice YungWkndRenderer address
+     */
+    IYungWkndRenderer public renderer;
+
+    /**
+     * @notice Store the block hash for every minted token batch
+     */
+    mapping(uint256 => SeedDetails) private _startTokenIdToSeedDetails;
+
+    /**
+     * @notice Store the first token id of each minted batch
+     */
+    uint256[] private _startTokenIds;
+
+    /**
+     * @notice Store the image previews base uri
+     */
+    string private _previewsBaseUri;
+
+    /**
+     * @notice Emit when YungWkndRenderer is updated
+     */
+    event RendererUpdated(address indexed newRenderer);
+
+    /* solhint-disable not-rely-on-block-hash */
+    /**
+     * @notice See {IERC1155GeneralMint-mintOneToOneRecipient}
+     * @dev Update YungWknd mint details
+     */
+    function mintOneToOneRecipient(address recipient) external override onlyMinter nonReentrant returns (uint256) {
+        if (_mintFrozen == 1) {
+            _revert(MintFrozen.selector);
+        }
+
+        uint256 tempSupply = _tokenCount;
+        _requireLimitSupply(tempSupply);
+
+        _mint(recipient, tempSupply, 1, "");
+
+        _startTokenIds.push(tempSupply);
+        _startTokenIdToSeedDetails[tempSupply] = SeedDetails(blockhash(block.number - 1), block.timestamp);
+
+        return tempSupply;
+    }
+
+    /* solhint-enable not-rely-on-block-hash */
+    /**
+     * @notice See {IERC1155GeneralMint-mintExistingOneToOneRecipient}
+     * @dev Update YungWknd mint details
+     */
+    function mintExistingOneToOneRecipient(address recipient, uint256 tokenId) external override onlyMinter nonReentrant returns (uint256) {
+        if (_mintFrozen == 1) {
+            _revert(MintFrozen.selector);
+        }
+
+        _requireLimitSupply(tokenId);
+
+        _mint(recipient, tokenId, 1, "");
+
+        // _startTokenIds.push(tempSupply);
+        // _startTokenIdToSeedDetails[tempSupply] = SeedDetails(blockhash(block.number - 1), block.timestamp);
+
+        return 1;
+    }
+
+    /* solhint-enable not-rely-on-block-hash */
+    /**
+     * @notice See {IERC1155GeneralMint-mintSeedToOneRecipient}
+     * @dev Update YungWknd mint details
+     */
+    function mintSeedToOneRecipient(address recipient, bytes32 seed) external override onlyMinter nonReentrant returns (uint256) {
+        if (_mintFrozen == 1) {
+            _revert(MintFrozen.selector);
+        }
+
+        uint256 tempSupply = _tokenCount;
+        _requireLimitSupply(tempSupply);
+
+        _mint(recipient, tempSupply, 1, "");
+
+        _startTokenIds.push(tempSupply);
+        _startTokenIdToSeedDetails[tempSupply] = SeedDetails(blockhash(block.number - 1), block.timestamp);
```

Comment:
One question I would have here is - 

Can you go more in depth on `SeedDetails` and how that hooks up to the art generation off-chain? If I pass in a `bytes32 seed` to `mintSeedToOneRecipient` then would this line be like -

```
_startTokenIdToSeedDetails[tempSupply] = SeedDetails(seed, block.timestamp);
```